client/src/ui/Form.jsx

```diff
@@ -326,6 +351,103 @@ export default class Form extends Component {
     }
   }
 
+  getGradeOptions() {
+      const gradeOptions = [];
+      this.gradeRecieved = ["A+", "A", "A-", "B+", "B", "B-", "C+", "C", "C-", "D+", "D", "D-", "F"];
```

Comment:
I might be missing this somewhere, but where is `gradeReceived` previously declared? And, do you access it outside of this function?